engine/execution/state/delta/view.go

```diff
@@ -28,7 +29,7 @@ type View struct {
 
 type Snapshot struct {
 	Delta Delta
-	Reads []flow.RegisterID
+	Reads map[string]flow.RegisterID
```

Comment:
Reads were populated from map anyway, so it didn't make sense to use list here and keep a false sense of ordering

Reply:
I remember @turbolent  posted a while ago that is more performant to define a key struct type (e.g. registerID in this case) and use that as a key instead of constructing a string of it.

Reply:
not sure I get this - you can't use any struct as a key

Reply:
This comment might be totally unrelated: just making sure that removing the ordering here won't change anything with the reads ordering in spock secrets.

Reply:
> This comment might be totally unrelated: just making sure that removing the ordering here won't change anything with the reads ordering in spock secrets.
It shouldn't this slice was populated from a map anyway